### ISSUE TITLE
lec7. 머신러닝 Tip

### ISSUE THREAD

**Issue by sculove** [studye/yeahEE#8]
### 1. Learning rate 조절
Gradient descent의 learning_rate

#### 어떻게 잘 정할 것인가? (한 발자국)
- 큰 값이면? 절대 안돼. 이때 overshooting이 발생함 (cost값이 숫자가 아닌값이 나오기도 함)
- 굉장히 작은 값이면? 절라 오래걸림.

#### 어떻게 하는가?
- 데이터, 환경에 따라 다름
- 보통 0.01부터 시작해서 조절해야한다. 

### 2. data 선처리
값들이 굉장히 큰 차이가 난다면?
등고선이 길쭉한 형태로 봐뀔꺼다.
![2017-10-26 9 57 31](https://user-images.githubusercontent.com/1383952/32054227-b4af2b0c-ba98-11e7-984f-a51085220602.png)

노멀라이즈 해야한다.
- zero-centered data : 데이터의 중심이 0으로 가도록 바꿈
- normalized data : 특정 범위로 들어가게 하는 식으로 데이터를 변경함.
![2017-10-26 9 57 59](https://user-images.githubusercontent.com/1383952/32054244-c3c2c716-ba98-11e7-9521-f46bd48b2041.png)

### 3. overfitting
학습데이터에만 너무 잘맞는 모델이 만들어짐.
실제 데이터와는 안맞는 경우가 생김.
![2017-10-26 9 58 28](https://user-images.githubusercontent.com/1383952/32054268-d4c363c2-ba98-11e7-831e-6fd8c8eef517.png)

#### 어떻게 할것인가?
- 트레닝 데이터가 많이 가지고 잇는 경우.
- 기능의 수를 줄이는 것
- regularization을 한다.
 : 우리가 가진 값을 너무 큰값을 갖지말자? 구부르지 말고 피자.
 코스트 함수에 텀(regulazation strength)을 추가해줌. 각각 엘리먼트를 제곱해서 합치는 것에 상수를 주자.
![2017-10-26 9 59 09](https://user-images.githubusercontent.com/1383952/32054289-eb330996-ba98-11e7-83fd-c6baabd43cb9.png)
